include/primitive_solver.hpp

```diff
@@ -366,6 +366,13 @@ inline SolverResult PrimitiveSolver<EOSPolicy, ErrorPolicy>::ConToPrim(Real prim
     solver_result.error = Error::CONS_FLOOR;
     return solver_result;
   }
+  // If a floor is applied or Y is adjusted, we need to propagate the changes back to
+  // DYe.
+  if (floored || Y_adjusted) {
+    for (int s = 0; s < n_species; s++) {
+      cons[IYD + s] = cons[IYD]*Y[s];
```

Comment:
Should use `D` instead of `cons[IYD]` as this may have been changed by `ApplyConservedFloor`.

Reply:
Done.